### DISCUSSION TITLE
Returning a single field from Model in Sequelize with findOne/findAll

### DISCUSSION THREAD

**Discussion by guenonjerome** [sequelize/sequelize#17270] [Help & Questions]
Hello! Is there a way to retrieve from database, using the last version of Sequelize without the overhead of loading a model(and of course, more elegant than the following)?

`const product = await Product.findOne({ where: { id: id }, attributes: ['hash'], transaction: tr }); return product?.hash;`

I know that if I pass { raw: true }, the overhead is supposed to be minimal, but even so, the code above is not very idiomatic nor elegant for the purpose of getting a value from just one field, it's too verbose as well.

Please don't tell me to do a raw SQL query, I already know that can be done, but I'm choosing not to, because of reasons I wont go in to details on here.

Look for instance how elegantly Sequelize handles this:

`return await
    Product.max<number, Product>('level',
        {
            where: {
                hash: hash
            },
            transaction: tr
        }
    );`

I was looking for something similar, but so far have been unsuccessful. In the world of Java that is so simple and beautiful, as well as available since way way way back.

Thanks.

**Comment by ephys** [comment]
That is currently the only way, sorry

> In the world of Java that is so simple and beautiful, as well as available since way way way back.

Do you have links / examples? I'm curious about what it looks like

**Reply by guenonjerome** [reply]
Very much like the example I gave about the max() method from Sequelize, it would depend on what library or tech you were using underneath. Hibernate had something similar almost 20 years ago, it was done using HQL, when the JPA standard came about, you would use JPAQL + generics to infer the type it would return. Nowadays you have Spring Data, that would look at the return type of the method at a repository interface and infer the type you'd be returning from your query(not "infer" per say, but expect. So much so, that it can cause exceptions at runtime if your method returns an Integer and your JPAQL returns a VARCHAR for instance).

**Reply by ephys** [reply]
I'm still a bit confused about the feature you're requesting, which is why code examples or links to code examples to how other technologies handle it would be very useful

Is it a method dedicated so selecting a single attribute? Something like

```ts
await User.selectSingleAttribute('hash', { where: { id: id }, transaction: tr });
```

Is it something more like what [Kysely](https://kysely.dev/) does with raw queries being automatically well typed? (This is something I would like to get to, eventually)
